cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/SelfAwareJoinDecider.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2017-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.management.cluster.bootstrap
+
+import akka.actor.ActorSystem
+import akka.discovery.SimpleServiceDiscovery.ResolvedTarget
+import akka.event.Logging
+
+import scala.concurrent.Await
+import scala.util.Try
+
+/**
+ * Class for further behavior in a [[akka.management.cluster.bootstrap.JoinDecider]]
+ * leveraging self host logic.
+ */
+private[bootstrap] abstract class SelfAwareJoinDecider(system: ActorSystem, settings: ClusterBootstrapSettings)
+    extends JoinDecider {
+
+  protected val log = Logging(system, getClass)
+
+  private def discoveryTimeout = settings.contactPointDiscovery.selfDiscoveryTimeout
+
+  /** Returns the current `selfContactPoints` as a String for logging, e.g. [127.0.0.1:64714]. */
+  protected def contactPointsString(contactPoints: Set[(String, Int)]): String =
+    contactPoints.map(_.productIterator.mkString(":")).mkString("[", ",", "]")
+
+  /**
+   * The value `ClusterBootstrap(system).selfContactPoints` is set prior to HTTP binding,
+   * during [[akka.management.AkkaManagement.start()]], hence we accept blocking on
+   * this initialization.
+   */
+  private[bootstrap] def selfContactPoints: Set[(String, Int)] =
+    Try(Await.result(ClusterBootstrap(system).selfContactPoints, discoveryTimeout))
+      .getOrElse(throw new IllegalStateException(
+            "'Bootstrap.selfContactPoint' was NOT set, but is required for the bootstrap to work " +
```

Comment:
I can't think of a usecase where these routes aren't hosted in AkkaManagement. 

I had to look at the code to see what this meant, and what the user would need to do if using the bootstrap routes outside of Akka Management is call `ClusterBootstrap(system).setSelfContactPoint()` but it is a `private[akka]` method so really that'll only be us.

Reply:
This is relocated code which was duplicated (oddly) in the lowest address decider - so I removed the duplication and relocated the function to here.

Reply:
Ah no sorry let me explain better. The only reason this is an Await is because we don't have the selfContactPoints until Akka Management calls https://github.com/akka/akka-management/blob/master/cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/ClusterBootstrap.scala#L106  via https://github.com/akka/akka-management/blob/master/cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/ClusterBootstrap.scala#L67 which it does before bootstrap is allowed to start so this should never actually have to wait. If this times out then it means there's a bug in Akka Management rather than it taking a long time. 

So in that case I don't think exposing a timeout to a user is helpful because it should either be set on startup or never (i.e. if we have a bug).

Reply:
@chbatey ACK, updating :)